### PR TITLE
Remove served and simplify flush

### PR DESCRIPTION
**Pull request by oleggrishechkin**
related to https://github.com/maverick-js/scheduler/issues/1

@mihar-22 This is what I mean. This pr should improve perfomance. Set add and delete is fast, but this is an overhead.

I use `queue = new Set()` because I believe this is faster than `queue.clear()`. But maybe I'm wrong.

Also, API doesn't changed, so this can be a patch.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -51,15 +49,11 @@ export function createScheduler(): Scheduler {
...
-        served.add(task);
-        queue.delete(task);
-      }
-    } while (queue.size > 0);
+    for (const task of queue) {
+      task();
+    }
 
-    served.clear();
+    queue = new Set();
```
Comment (mihar-22) [inline src/index.ts]: I'd love to know if this is true because `.clear()` is extremely efficient as far as I know.

Comment (oleggrishechkin) [reply src/index.ts]: Interesting
I chacked FF, Safari and Chrome - only in chrome creating new instance is faster (~5x)

In Safary .clear() faster (up to 3x) and in FF clear much much faster.

So, I'll keep clear()

Comment (oleggrishechkin) [reply src/index.ts]: @mihar-22 I used .clear() now

Comment (mihar-22) [reply src/index.ts]: Thanks heaps for testing that @oleggrishechkin!

**Pull request merged**